### PR TITLE
[Tradfri] Fix bug in color space conversion

### PR DESCRIPTION
**Pull request by hreichert**
Ensure that calculated RGB values are not negative.
Added new unit test case.

Fixes: #4349

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriColor.java [hunk 1]
```diff
@@ -124,9 +124,9 @@ public static TradfriColor fromCie(int xyX, int xyY, int xyBrightness) {
         // green = green <= 0.0031308 ? 12.92 * green : (1.0 + 0.055) * Math.pow(green, (1.0 / 2.4)) - 0.055;
         // blue = blue <= 0.0031308 ? 12.92 * blue : (1.0 + 0.055) * Math.pow(blue, (1.0 / 2.4)) - 0.055;
 
-        int redRounded = (int) Math.round(red * 255.0);
-        int greenRounded = (int) Math.round(green * 255.0);
-        int blueRounded = (int) Math.round(blue * 255.0);
+        int redRounded = ensureRgbRange((int) Math.round(red * 255.0));
```
Comment (triller-telekom) [inline extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriColor.java]: I wonder if this is the right place for the fix. Shouldn't we make sure that red/green/blue is not negative and not above 1? At least the later is already checked in lines 108-120 so you would only need to check for negative numbers.

Comment (hreichert) [reply extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriColor.java]: Thanks for your comment, that would be indeed better.
I'm going to change this and rebase on master, as there are now duplicate method names in the tests I guess.

Comment (hreichert) [reply extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriColor.java]: Done.

**Comment by hreichert** [comment]
Updated PR with suggested change from @triller-telekom

**Comment by sjka** [comment]
Thanks!

**Pull request merged**